### DISCUSSION TITLE
Unleash Edge with a Redis Cluster

### DISCUSSION THREAD

**Discussion by Ganthi753** [Unleash/unleash#5449] [Q&A]
I am trying to get Unleash Edge running with a Redis Cluster. I cannot find any documentation on using a cluster, and when I try to specify a single Redis Node from from the cluster I get repeated errors from Unleash Edge that data cannot be persisted because it has moved like this:

WARN
unleash_edge::persistence
Could not persist features: PersistenceError("Error connecting to Redis: An error was signalled by the server - Moved: 8400 {second node in cluster}

Is there something I am missing, or does Unleash Edge not support a Redis Cluster?

If this is something where Unleash Edge does not support a Redis Cluster, I am wondering if I misunderstand the purpose of using Redis with Unleash Edge. Does Unleash Edge use Redis to store a cache of feature flags so that multiple Unleash Edge nodes can rely on the same Redis back end for cached flags, or is Redis just used at the time that Unleash restarts to reload flags that are already in Redis, and each Unleash Edge node keeps a local cache of flag that are refreshed when FEATURES_REFRESH_INTERVAL_SECONDS timeout expires?

Thanks for any direction you can provide!
Jeremie Roberts

**Comment by FredrikOseberg** [comment]
> I am trying to get Unleash Edge running with a Redis Cluster. I cannot find any documentation on using a cluster, and when I try to specify a single Redis Node from from the cluster I get repeated errors from Unleash Edge that data cannot be persisted because it has moved like this:
> 
> WARN unleash_edge::persistence Could not persist features: PersistenceError("Error connecting to Redis: An error was signalled by the server - Moved: 8400 {second node in cluster}
> 
> Is there something I am missing, or does Unleash Edge not support a Redis Cluster?
> 
> If this is something where Unleash Edge does not support a Redis Cluster, I am wondering if I misunderstand the purpose of using Redis with Unleash Edge. Does Unleash Edge use Redis to store a cache of feature flags so that multiple Unleash Edge nodes can rely on the same Redis back end for cached flags, or is Redis just used at the time that Unleash restarts to reload flags that are already in Redis, and each Unleash Edge node keeps a local cache of flag that are refreshed when FEATURES_REFRESH_INTERVAL_SECONDS timeout expires?
> 
> Thanks for any direction you can provide! Jeremie Roberts

@Ganthi753 

Hello Jeremie

Thanks for using Unleash! You are correct, currently we don't support a redis cluster to be used with Unleash-Edge. Currently Redis works as a backup mechanism for edge which it will use until it can reach back up to upstream. Edge will use the persistent cache in case the upstream server is not available, and periodically sync its state to this persistent cache when configured. There is some more information about how this works in this section of the documentation, right after the daisy chaining section: https://github.com/Unleash/unleash-edge#modes

I'm guessing that the node moved error is due to the node moving internally in the cluster?

Hopefully this provides some insight into how edge works with Redis.

All the best,
Fredrik Strand Oseberg

**Comment by gardleopard** [comment]
This ticket is being closed due to inactivity. Please feel free to reopen the ticket.
